src/lambda/config/templates.ts

```diff
@@ -395,3 +398,67 @@ export class TemplatesConfigPopulator {
         return this
     }
 }
+
+/**
+ * Returns a previously-configured event payload if the following are true (`undefined` if false):
+ * * old `.aws/templates.json` file exists and is valid JSON
+ * * handler exists in template registry
+ * * handler also exists in `.aws/templates.json`
+ * @param workspaceFolder Current workspace folder
+ * @param handler Function's handler
+ * @param samTemplate Originating SAM Template
+ */
+export async function getExistingConfiguration(
+    workspaceFolder: vscode.WorkspaceFolder,
+    handler: string,
+    samTemplate: vscode.Uri,
+    registry: CloudFormationTemplateRegistry = CloudFormationTemplateRegistry.getRegistry()
+): Promise<
+    | {
```

Comment:
I had one here originally, removed at @justinmk3 's request. We should standardize on this.

Reply:
They aren't needed.

Reply:
Sure, there's definitely pros and cons. Certainly we should use an interface for any type that's repeated in method signatures to keep it DRY. 

In this case of a single method, it's less important. I would argue, however, that the long method signature makes it noisier to read.

You are correct that assigning an explicit type leads to an extra level of indirection while reading code. However, in practice, I've found that it's better to make it easier to skim past code (extra indirection) than to make it easier to get the deep details (in-lining stuff). 

Reading code at a higher level of abstraction is more common than needing to dig into the details. Additionally, it helps you split out functionality. And you can always dive into the details if you need to. IDE assistance, tooltips, etc. can inline code for you. But if you inline yourself, IDE assistance can't factor out your code to let you read it at a higher level of abstraction.

> What would the doc on the type say, that this anonymous type doesn't already convey?

I see plenty of opportunity to attach doc to the outputs here. Stuff like `dockerNetwork` could use some explanation as to the values. A benefit of explicit interfaces with doc is that the doc is tied to the interface rather than the method where the inline interface is defined.

Reply:
> Certainly we should use an interface for any type that's repeated in method signatures to keep it DRY.

Yes. A few core common data structures which are used in many methods, is powerful and high-leverage. But we don't do that. We have a habit of creating interfaces for every method, and that is noisy. This is such a case.

Reply:
Yeah for something used once like `{ isSuccess: boolean }` I'm on board. Especially if the functions are private. I'm not sure about a richer 4-property response on an exported function though.